templates/flavors/flatcar/patches/kubeadm-controlplane.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+kind: KubeadmControlPlane
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  kubeadmConfigSpec:
+    diskSetup:
+      filesystems:
+      - device: /dev/disk/azure/scsi1/lun0
+        extraOpts:
+        - -E
+        - lazy_itable_init=1,lazy_journal_init=1
+        filesystem: ext4
+        label: etcd_disk
+      partitions:
+      - device: /dev/disk/azure/scsi1/lun0
+        layout: true
+        overwrite: false
+    initConfiguration:
+      nodeRegistration:
+        name: "@@HOSTNAME@@"
+    joinConfiguration:
+      nodeRegistration:
+        name: "@@HOSTNAME@@"
+    mounts:
+    - [etcd_disk, /var/lib/etcddisk]
+    format: ignition
+    ignition:
+      containerLinuxConfig:
+        additionalConfig: |
+          systemd:
+            units:
+            - name: kubeadm.service
+              dropins:
+              - name: 10-flatcar.conf
+                contents: |
+                  [Unit]
+                  After=oem-cloudinit.service
+    postKubeadmCommands: []
+    preKubeadmCommands:
+    # Hostname for controller nodes is not really required, but it is added for consistency with worker nodes.
+    - sed -i "s/@@HOSTNAME@@/$(curl -s -H Metadata:true --noproxy '*' 'http://169.254.169.254/metadata/instance?api-version=2020-09-01' | jq -r .compute.name)/g" /etc/kubeadm.yml
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+spec:
+  template:
+    spec:
+      image:
+        sharedGallery:
+          gallery: ClusterAPI
+          name: capi-flatcar-stable-2905.2.3
+          offer: flatcar-container-linux-free
+          publisher: kinvolk
+          resourceGroup: cluster-api-images
+          sku: stable
+          subscriptionID: ${AZURE_SUBSCRIPTION_ID}
+          version: 0.3.1632385693
```

Comment:
Missing newline at the end of the file. It's nice to have it, so when you do `cat ...`, it does not mess up your prompt :)

Reply:
Should be there now